radicle-cli/src/commands/inspect.rs

```diff
@@ -0,0 +1,222 @@
+#![allow(clippy::or_fun_call)]
+use std::ffi::OsString;
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+use std::str::FromStr;
+
+use anyhow::{anyhow, Context as _};
+use chrono::prelude::*;
+use json_color::{Color, Colorizer};
+
+use radicle::git;
+use radicle::identity::project::{Doc, IdentityError, Untrusted};
+use radicle::identity::Id;
+use radicle::storage::{ReadRepository, ReadStorage, WriteStorage};
+
+use crate::terminal as term;
+use crate::terminal::args::{Args, Error, Help};
+
+pub const HELP: Help = Help {
+    name: "inspect",
+    description: "Inspect a radicle identity or project directory",
+    version: env!("CARGO_PKG_VERSION"),
+    usage: r#"
+Usage
+
+    rad inspect <path> [<option>...]
+    rad inspect <id>   [<option>...]
+    rad inspect
+
+    Inspects the given path or ID. If neither is specified,
+    the current project is inspected.
+
+Options
+
+    --id-only   Return the ID in simplified form
+    --payload   Inspect the object's payload
+    --refs      Inspect the object's refs on the local device (requires `tree`)
+    --history   Show object's history
+    --help      Print help
+"#,
+};
+
+#[derive(Default, Debug, Eq, PartialEq)]
+pub struct Options {
+    pub path: Option<PathBuf>,
+    pub id: Option<Id>,
+    pub refs: bool,
+    pub payload: bool,
+    pub history: bool,
+    pub id_only: bool,
+}
+
+impl Args for Options {
+    fn from_args(args: Vec<OsString>) -> anyhow::Result<(Self, Vec<OsString>)> {
+        use lexopt::prelude::*;
+
+        let mut parser = lexopt::Parser::from_args(args);
+        let mut path: Option<PathBuf> = None;
+        let mut id: Option<Id> = None;
+        let mut refs = false;
+        let mut payload = false;
+        let mut history = false;
+        let mut id_only = false;
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Long("help") => {
+                    return Err(Error::Help.into());
+                }
+                Long("refs") => {
+                    refs = true;
+                }
+                Long("payload") => {
+                    payload = true;
+                }
+                Long("history") => {
+                    history = true;
+                }
+                Long("id-only") => {
+                    id_only = true;
+                }
+                Value(val) if path.is_none() && id.is_none() => {
+                    let val = val.to_string_lossy();
+
+                    if let Ok(val) = Id::from_str(&val) {
+                        id = Some(val);
+                    } else if val.starts_with("rad:git:") {
```

Comment:
is `rad:git:` still applicable? If not, how to fix this? separate flags for path & id?

Reply:
Hmm can just remove the branch? I guess maybe we should check if the path exists instead?

Reply:
[diff](https://github.com/radicle-dev/heartwood/compare/34de4696e1e70c63b31d4c7cde6de51c24a72371..86f46c16b9ea02c83094179ff3338631d94a608f#diff-9581cd238b2996c77fa73aa7caa180f7f668ac36ff3f182f1aa8f02e69a4d2e1L87-L88)